source/elements/oneDPL/source/parallel_api/iterators.rst

```diff
@@ -301,3 +339,108 @@ operation were applied to each of these iterators. The types ``T`` within the te
 using the set of source iterators provided.
+
+.. _iterators-device-accessible:
+
+Customization For User Defined Iterators
+++++++++++++++++++++++++++++++++++++++++
+
+oneDPL provides a mechanism to indicate whether custom iterators are "device accessible content iterators" by defining
+an Argument-Dependent Lookup (ADL) based customization point, ``is_onedpl_device_accessible_content_iterator``. oneDPL
+also defines a public trait, ``is_device_accessible_content_iterator[_v]`` to indicate whether an iterator is a "device
+accessible content iterator".
+
+oneDPL queries this information at compile time to determine how to handle iterator types when they are passed to
+algorithms with a ``device_policy`` to avoid unnecessary data movement.
+
+ADL Customization Point: ``is_onedpl_device_accessible_content_iterator``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A free function ``is_onedpl_device_accessible_content_iterator(IteratorT)`` may be defined, which accepts an argument
+of type ``IteratorT`` and returns a type with the characteristics of ``std::true_type`` if ``IteratorT`` refers to
+"device accessible content", or otherwise returns a type with the characteristics of ``std::false_type``. The function
+must be discoverable by ADL.
+
+The function ``is_onedpl_device_accessible_content_iterator`` may be used by oneDPL to determine if the iterator refers
+to "device accessible content" by interrogating its return type at compile-time only. Overloads may be provided as
+forward declarations only, without a body defined. ADL is used to determine which function overload to use according to
+the rules in the `C++ Standard`_. Therefore, derived iterator types without an overload for their exact type will match
+their most specific base iterator type if such an overload exists.
+
+The default implementation of ``is_onedpl_device_accessible_content_iterator`` marks the following iterators as to
+"device accessible content iterators":
+* Pointers (to handle USM pointers)
+* ``std::reverse_iterator<IteratorT>`` when ``IteratorT`` refers to "device accessible content"
+
+
+
+Public Trait: ``oneapi::dpl::is_device_accessible_content_iterator[_v]``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following class template and variable template are defined in ``<oneapi/dpl/iterator>`` inside the namespace
+``oneapi::dpl``:
+
+.. code:: cpp
+
+    template <typename T>
+    struct is_device_accessible_content_iterator{ /* see below */ };
+
+    template <typename T>
+    inline constexpr bool is_device_accessible_content_iterator_v = is_device_accessible_content_iterator<T>::value;
+
+``template <typename T> oneapi::dpl::is_device_accessible_content_iterator<T>`` evaluates to a type with the 
+characteristics of ``std::true_type`` if ``T``  refers to "device accessible content", otherwise it evaluates to a type
+with the characteristics of ``std::false_type``.
```

Comment:
I would still recommend some improvements
```suggestion
``is_device_accessible_content_iterator`` has characteristics of ``std::true_type`` if ``T``  refers to "device accessible content", otherwise it has characteristics of ``std::false_type``.
```
If you want to keep `template <typename T>` at the beginning, that's probably fine (though, looks redundant) but please at least remove `<T>` after class name. Also, "evaluates to a type" sounds a little bit clunky to me.

Reply:
fixed.